core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1647,6 +1645,24 @@ class SnappySession(@transient private val sc: SparkContext,
 
   def setPreparedQuery(preparePhase: Boolean, paramSet: Option[ParameterValueSet]): Unit =
     sessionState.setPreparedQuery(preparePhase, paramSet)
+
+  /**
+   * Callback to execute code when a particular property is set.
+   * @param key
+   * @param value
+   */
+  def confCallback(key: String, value: String): Unit = {
+    key match {
+      case "spark.scheduler.pool" =>
+        // set the scheduler pool if the pool exists else throw an exception. 
+        if (sparkContext.getAllPools.exists(_.name == value)) {
+          sparkContext.setLocalProperty(key, value)
+        } else {
+          throw new IllegalArgumentException(s"Invalid Pool $value")
+        }
+      case _ =>
+    }
+  }
```

Comment:
>>Weren't you arguing that users will continue to use SparkContext.setLocalProperty in jobs and smart connector??

No. I was always talking about the embedded and smart connector. Not about jdbc versus scala code. 

>> Exactly. And as per your own stance earlier, for other cases users will continue to use SnappyContext.setLocalProperty, no? Or did I misunderstand what you were saying earlier

Yes, I think you misunderstood. Check the PR description. SparkContext.setLocalProperty API creates a new pool if it does not already exist. We cannot allow that. 

>> But we must use the SnappyConf way so that then the same property can be used for both Spark Hive JDBC/ODBC (when we add SnappySession support in that) as well as SnappyData JDBC/ODBC.

You mean same property name should be added to SnappyConf? I can do that as far as SnappyConf allows it.